tests/topotests/ospfv3_basic_functionality/test_ospfv3_authentication.py

```diff
@@ -0,0 +1,870 @@
+# for any purpose with or without fee is hereby granted, provided
+# that the above copyright notice and this permission notice appear
+# in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND VMWARE DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL VMWARE BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY
+# DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS
+# ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE
+# OF THIS SOFTWARE.
+#
+
+
+"""OSPF Basic Functionality Automation."""
+import os
+import sys
+import time
+import pytest
+from time import sleep
+from copy import deepcopy
+import json
+from lib.topotest import frr_unicode
+
+pytestmark = pytest.mark.ospf6d
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+sys.path.append(os.path.join(CWD, "../lib/"))
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from mininet.topo import Topo
+from lib.topogen import Topogen, get_topogen
+
+# Import topoJson from lib, to create topology and initial configuration
+from lib.common_config import (
+    start_topology,
+    write_test_header,
+    write_test_footer,
+    reset_config_on_routers,
+    step,
+    shutdown_bringup_interface,
+    topo_daemons,
+)
+from lib.topolog import logger
+from lib.topojson import build_topo_from_json, build_config_from_json
+from lib.ospf import verify_ospf6_neighbor, config_ospf6_interface, clear_ospf
+from ipaddress import IPv4Address
+
+# Global variables
+topo = None
+# Reading the data from JSON File for topology creation
+jsonFile = "{}/ospfv3_authentication.json".format(CWD)
+try:
+    with open(jsonFile, "r") as topoJson:
+        topo = json.load(topoJson)
+except IOError:
+    assert False, "Could not read file {}".format(jsonFile)
+"""
+TOPOOLOGY =
+      Please view in a fixed-width font such as Courier.
+      +---+  A1       +---+
+      +R1 +------------+R2 |
+      +-+-+-           +--++
+        |  --        --  |
+        |    -- A0 --    |
+      A0|      ----      |
+        |      ----      | A2
+        |    --    --    |
+        |  --        --  |
+      +-+-+-            +-+-+
+      +R0 +-------------+R3 |
+      +---+     A3     +---+
+
+TESTCASES =
+1.  OSPFv3 Authentication Trailer - Verify ospfv3 authentication trailer
+    using MD5 manual key configuration.
+2.  OSPFv3 Authentication Trailer - Verify ospfv3 authentication trailer
+    using HMAC-SHA-256 manual key configuration.
+3.  OSPFv3 Authentication Trailer - Verify ospfv3 authentication trailer
+    using MD5 keychain configuration.
+4.  OSPFv3 Authentication Trailer - Verify ospfv3 authentication trailer
+    using HMAC-SHA-256 keychain configuration.
```

Comment:
Please add a test like this:

```
1. start all ospf6d
2. ensure topology converged
3. make some ospf traffic, for example announce & withdraw 100 routes (purpose is to increase seqnum because OSPF packets will be exchanged)
4. restart ONE of the ospf6d (NOT all!)
5. check that topology converges again within a few seconds
```

If I am correct, it will fail because we do not store the seqnum in persistent storage like https://datatracker.ietf.org/doc/html/rfc7166#section-4.1.1 requires.  We cannot proceed without this since there is no recovery mechanism in the RFC.

Also refer to a few lines above the RFC link:

```
      OSPFv3 routers implementing this specification MUST use available
      mechanisms to preserve the sequence number's strictly increasing
      property for the deployed life of the OSPFv3 router (including
      cold restarts).  One mechanism for accomplishing this would be to
      use the high-order 32 bits of the sequence number as a wrap/boot
      count that is incremented anytime the OSPFv3 router loses its
      sequence number state.  Sequence number wrap is described in
      Section 4.1.1.
```

This MUST is not covered by your implementation, and it will break when ospf6d is restarted.

Reply:
Hi @eqvinox I have this test case tested. Since the interface creation in ospf is persistent until you have some configuration. old sequence number persists across ospf process restart. There is no special handling required for this.

Reply:
> Hi @eqvinox I have this test case tested. Since the interface creation in ospf is persistent until you have some configuration. old sequence number persists across ospf process restart. There is no special handling required for this.

I'm talking about restarting ospf6d, not reconfiguring the interface.

Reply:
> old sequence number persists across ospf process restart.

?????